src/services/codefixes/superFixes.ts

```diff
@@ -0,0 +1,53 @@
+﻿/* @internal */
+namespace ts.codeFix {
+    function getOpenBraceEnd(constructor: ConstructorDeclaration, sourceFile: SourceFile) {
+        // First token is the open curly, this is where we want to put the 'super' call.
+        return constructor.body.getFirstToken(sourceFile).getEnd();
+    }
+
+    registerCodeFix({
+        name: getLocaleSpecificMessage(Diagnostics.Add_missing_super_call),
+        errorCodes: ["TS2377"],
+        getTextChanges: (sourceFile: SourceFile, start: number, end: number) => {
+            const token = getTokenAtPosition(sourceFile, start);
+            Debug.assert(token.kind === SyntaxKind.ConstructorKeyword,"Failed to find the constructor.");
+
+            const newPosition = getOpenBraceEnd(<ConstructorDeclaration>token.parent, sourceFile);
+
+            return [{ newText: "super();", span: { start: newPosition, length: 0 } }];

```

Comment:
The managed side deals with formatting the affected code-range, since it know about the required indentation etc.